ledger/apply/stateproof.go

```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/algorand/go-algorand/config"
 	"github.com/algorand/go-algorand/data/basics"
 	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
```

Comment:
I don't believe moving this file to `ledger/internal` will be consistent with the current repo hierarchy for the following reasons:

1. Every other transaction applying logic is located under `ledger/apply`:
    - `ledger/apply/keyreg.go`
    - `ledger/apply/payment.go`
    - `ledger/apply/asset.go`
    - `ledger/apply/application.go`
    Applying `StateProof` transaction in a different location than every other transaction is not consistent.

2. The semi-circular dependency you've mentioned has been in this package regardless of our changes, and moving this file to the internal package would not solve that either way.

Reply:
The code structure says otherwise. 
This file is introducing a single interface: `func StateProof` and this is an exported function only because it is in the wrong package. No other reason (no other usage for it).
Having exported interfaces complicates the overcall cod structure. The more local things are, the more readable and understandable the code becomes.

Reply:
This is the same pattern used for payment, keyreg, asset transfer ,and app call transactions as well though.
Your logic is sound, but it can be applied to every other transaction I've listed, so singling out the StateProof transaction does not make sense in my opinion.

Either we refactor them all into `ledger/internal` and get rid of the package `ledger/apply`, or we stay consistent to the current chosen structure.
I'm not necessarily against refactoring that part of the code, but it seems out of scope for this feature.